lib/ansible/modules/system/lvol_thin.py

```diff
@@ -0,0 +1,918 @@
+    default: present
+    description:
+    - Control if the logical volume exists. If C(present) the C(size) option
+      is required.
+    required: false
+  type:
+    choices: [thin, thin-pool, normal]
+    default: normal
+    description:
+    - Type of logical volume, thin is thind lvol, thin-pool is thin pool
+      supplying thind lvol, normal is lvol without any type. To create thin
+      lvol, thin pool must already exist.
+  vg:
+    description:
+    - The volume group this logical volume is part of.
+    required: true
+notes:
+  - Filesystems on top of the volume are not resized.
+'''
+
+EXAMPLES = '''
+- name: Create a logical volume of 512m
+  lvol:
+    vg: firefly
+    lv: test
+    size: 512m
+
+- name: Create a logical volume of 512g
+  lvol:
+    vg: firefly
+    lv: test
+    size: 512g
+
+- name: Create a logical volume the size of all remaining space in the volume group
+  lvol:
+    vg: firefly
+    lv: test
+    size: 100%FREE
+
+- name: Create a logical volume with special options
+  lvol:
+    vg: firefly
+    lv: test
+    size: 512g
+    opts: "-r 16"
+
+- name: Extend the logical volume to 1024m
+  lvol:
+    vg: firefly
+    lv: test
+    size: 1024m
+
+- name: Extend the logical volume to take all remaining space of the PVs
+  lvol:
+    vg: firefly
+    lv: test
+    size: 100%PVS
+
+- name: Resize the logical volume to % of VG
+  lvol:
+    vg: firefly
+    lv: test
+    size: 80%VG
+    force: yes
+
+- name: Reduce the logical volume to 512m
+  lvol:
+    vg: firefly
+    lv: test
+    size: 512m
+    force: yes
+
+- name: Remove the logical volume
+  lvol:
+    vg: firefly
+    lv: test
+    state: absent
+    force: yes
+
+- name: Create thin pool logical volume of 512m
+  lvol:
+    vg: firefly
+    lv: testpool
+    size: 512m
+    type: thin-pool
+
+- name: Create thin lvol testthind with thin pool firefly/testpool, thin pool must already exist
+  lvol:
+    vg: firefly
+    lv: testthind
+    size: 1g
+    type: thin
+    pool: firefly/testpool
+
+- name: Extend thin pool to 2g
+  lvol:
+    vg: firefly
+    lv: testpool
+    size: 2g
+    type: thin-pool
+
+- name: Extend thin lvol to 5g
+  lvol:
+    vg: firefly
+    lv: testthind
+    size: 5g
+    type: thin
+    pool: firefly/testpool
+
+- name: Convert thin lvol to normal lvol
+  lvol:
+    vg: firefly
+    lv: testthind
+    size: 5g
+    type: normal
+    force: yes
+
+- name: Convert normal lvol to thin pool
+  lvol:
+    vg: firefly
+    lv: testthind
+    size: 5g
+    type: thin-pool
+    force: yes
+
+- name: Convert thin pool lvol to thin lvol, you should notice that firefly/testpool still exists
+  lvol:
+    vg: firefly
+    lv: testthind
+    size: 5g
+    type: thin
+    pool: firefly/testpool
+    force: yes
+
+- name: Remove thin pool lvol, this also changes thin lvol to normal
+  lvol:
+    vg: firefly
+    lv: testpool
+    size: 2g
+    type: thin-pool
+    state: absent
+    force: yes
+'''
+
+RETURN = ''' # '''
```

Comment:
I would think what useful information you could possibly return to the user to work with from the registered variable.